syntax = "proto2";

package smartmouse.msgs;

import "row_col_yaw.proto";
import "ignition/msgs/time.proto";

message RobotSimState {
    optional ignition.msgs.Time stamp = 1;
    optional WheelPhysicsState left_wheel = 2;
    optional WheelPhysicsState right_wheel = 3;
    optional RowColYaw p = 4; // meters
    optional RowColYaw v = 5; // meters/second
    optional RowColYaw a = 6; // meters/second^2
    optional double front_left_adc = 7;
    optional double front_right_adc = 8;
    optional double gerald_left_adc = 9;
    optional double gerald_right_adc = 10;
    optional double back_left_adc = 11;
    optional double back_right_adc = 12;
    optional double front_adc = 13;
    optional double front_left_m = 14;
    optional double front_right_m = 15;
    optional double gerald_left_m = 16;
    optional double gerald_right_m = 17;
    optional double back_left_m = 18;
    optional double back_right_m = 19;
    optional double front_m = 20;
}

message WheelPhysicsState {
    optional double theta = 1; // radians
    optional double omega = 2; // radians/second
    optional double alpha = 3;
    optional double current = 4; // amperes
}
